brigand/algorithms/transform.hpp

```diff
@@ -103,11 +110,11 @@ namespace brigand
   namespace lazy
   {
     template<typename Sequence1, typename OpSeq1, typename... OpSeq2>
-    using transform = typename detail::transform_selector<sizeof...(OpSeq2), Sequence1, OpSeq1, OpSeq2...>;
+    using transform = typename detail::transform<Sequence1, OpSeq1, OpSeq2...>;
   }
 
   // Main transform entry point
   template<typename Sequence1, typename OpSeq1, typename... OpSeq2>
-  using transform = typename detail::transform_selector<sizeof...(OpSeq2), Sequence1, OpSeq1, OpSeq2...>::type;
+  using transform = typename detail::transform<Sequence1, OpSeq1, OpSeq2...>::type;
```

Comment:
Although initially I was thinking otherwise, the matching with `n` is better.

```c++
template<class L> struct run;

template<class i1, class i2, class i3, class i4>
struct func
{ using type = int; };

template<class... Ts> struct run<brigand::list<Ts...>>
{
  using type = brigand::list<brigand::transform<
    brigand::range<int, 0, 1+Ts::value>,
    brigand::range<int, 0, 1+Ts::value>,
    brigand::range<int, 0, 1+Ts::value>,
    brigand::range<int, 0, 1+Ts::value>,
    brigand::quote<func>
  >...>;
};

using list = run<brigand::range<int, 0, SIZE>>::type;
```

Gcc:

Size = |      50     |     100     |      200    |     300
------ | ----------- | ----------- | ------------ | ------------
before | 0.14s - 36M | 0.33s - 66M | 0.98s - 160M | 2.17s - 298M
after  | 0.15s - 36M | 0.33s - 67M | 0.99s - 163M | 2.22s - 305M

Clang:

Size = |      50     |     100     |      200    |     300
------ | ----------- | ----------- | ------------ | ------------
before | 0.20s - 47M | 0.50s - 71M | 1.46s - 159M | 2.98s - 297M
after  | 0.21s - 48M | 0.49s - 74M | 1.51s - 168M | 3.19s - 318M